### PR TITLE
feat: New Vesktop icon

### PR DESCRIPTION
**Pull request by Covkie**
Note: We decided to go with a different icon further down in the thread. [Click here](https://github.com/Vencord/Vesktop/pull/865#issuecomment-3344297874) to jump there.

____

feedback welcome.

Also cleaned up icons and removed dynamically setting the icon (electron builder handles this).

raw svg: 
vesktop

### CODE REVIEW

**Comment by Mikilio** [comment]
I just randomly appeared here. I quite like the design, but the background colors trip me up a bit (inparticular the darker lighter square on the lighter circle. There is also a discussion about how the details actually look on small icons, so maybe some details can be removed in favor of making the outlines pop more using contrast. (In particular the amount of hairpins and the "VC" look very noisy on the composition.)
You split the colors of the background diagonally, but maybe you may want to keep them single color and have an Icon for light mode and one for dark mode.

Even without any changes I'd be happy to see this on my Vesktop client, though.

**Comment by x0rp01s0n** [comment]
I've noticed a recurring issue with Vesktop that many users, including myself, find frustrating: the forced use of the rainbow logo and dancing anime characters when opening the app. This choice feels out of place for many, and yet there's no option for a more neutral appearance, like the basic Discord logo.

From my perspective, it would be more user-friendly to at least offer a choice. Many people have voiced similar concerns, with numerous entries expressing a desire for change. Despite this, the development team seems to dismiss these opinions, often with responses like "you don't need to use Vesktop" or accusations of "homophobia." Such responses come across as unfair and don't address the core issue.

Because of this, I'm seriously considering moving away from Vesktop. It’s not about personal judgment but simply about wanting to use an app every day without unnecessary visual elements that don't resonate with me or other users. It would be great if the devs took this feedback seriously and considered offering more customization options to cater to different preferences.

**Comment by Pukimaa** [comment]
> I've noticed a recurring issue with Vesktop that many users, including myself, find frustrating: the forced use of the rainbow logo and dancing anime characters when opening the app. This choice feels out of place for many, and yet there's no option for a more neutral appearance, like the basic Discord logo.
> 
> From my perspective, it would be more user-friendly to at least offer a choice. Many people have voiced similar concerns, with numerous entries expressing a desire for change. Despite this, the development team seems to dismiss these opinions, often with responses like "you don't need to use Vesktop" or accusations of "homophobia." Such responses come across as unfair and don't address the core issue.
> 
> Because of this, I'm seriously considering moving away from Vesktop. It’s not about personal judgment but simply about wanting to use an app every day without unnecessary visual elements that don't resonate with me or other users. It would be great if the devs took this feedback seriously and considered offering more customization options to cater to different preferences.

I totally agree. I'm very supportive regarding LGBTQIA+, yet I'd like to have a more "low-key" feel when opening Discord. It should feel seamless in my opinion. So I'd like to have regular Discord branding used.
Also I really like this icon too, good job ^^

**Comment by Mikilio** [comment]
So what are the steps to merging this? Will there be a vote, or who gets to decide? With these, it's always hard to call it a objective improvement.

**Comment by dphaldes** [comment]
Could the same icon be used for the system tray (atleast on linux) ?

**Comment by codeklepto** [comment]
Can we at least get a `--weeb=no` argument for initializing the use of icons for those of us over the age of 14?

**Comment by Vendicated** [comment]
you have an ai generated avatar I don't think you have room to talk

**Comment by nnra6864** [comment]
> Can we at least get a `--weeb=no` argument for initializing the use of icons for those of us over the age of 14?

Absolutely agreed...

**Comment by Casuallynoted** [comment]
I put together a liquid glass version of the icon in Icon Composer in the event we're looking to have support for that in macOS 26:

[Vencord.zip](https://github.com/user-attachments/files/20681333/Vencord.zip)

Screenshot 2025-06-10 at 7 34 33 PM
Screenshot 2025-06-10 at 7 34 51 PM
Screenshot 2025-06-10 at 7 35 08 PM

**Comment by chrrybmb64** [comment]
It should be mentioned that some icon packs on linux have icons for current Vesktop, so if this gets merged the new icon will not be shown on some people's systems.

Personally I don't think there's anything wrong with the current icon, but it's not my project, so if it changes I guess I'll just get used to it.

**Comment by nitram3** [comment]
Not sure if this is the place to post this, feel free to delete it, if it's not.
Would it be possible to add a custom icon feature? So each user could change it if they wanted to? Personally I would like to use the discord icon. Well the old one, as of today. :(

**Comment by Vendicated** [comment]
We (I?) decided that we won't be using this one (as much as I love it) because it doesn't really convey the function/purpose of the app and isn't really neutral (i.e. something everyone would like having on their desktop)

___

https://github.com/wingio came up with this design instead, which I'm currently considering moving forward with

image

image

**Comment by claration** [comment]
I like this icon very much but the mac icon that was provided in the recent changes feels very lackluster. 
image

I decided to make one that should fit the feel and style of recent macos versions (Tahoe) while also having it convert appropriately for light/dark/glass appearances. 

image
image
image



[vc.icns.zip](https://github.com/user-attachments/files/22607287/vc.icns.zip)

This file will also work on lower versions, just so people know.

**Comment by wingio** [comment]
does the more saturated pink not look as good?

**Comment by claration** [comment]
> does the more saturated pink not work as well?

@wingio you can be the judge of that 

image

**Comment by wingio** [comment]
> > does the more saturated pink not work as well?
> 
> @wingio you can be the judge of that 
> 
> image
> 

i think that works

**Comment by claration** [comment]
[vc.icns.zip](https://github.com/user-attachments/files/22607287/vc.icns.zip)

**Comment by claration** [comment]
When it comes to the tray icon we can probably use some variation of wings icon but without the chat bubble

tray

**Comment by wingio** [comment]
oh in the discord I posted a monochrome version

**Comment by claration** [comment]
any would work, most importantly should be clearly recognizable within the tray

**Comment by Covkie** [comment]
@khcrysalis I made the icon currently included with the idea that it would be for older macos versions while yours would be for macos 26+ with the new format. 
> This file will also work on lower versions, just so people know.

If it does work for lower versions & it's okay for the liquid glass style icon to be used on older macos ill swap it

Can you update the icon I tweaked the VC in the latest push. 
Also is the icns file you're sharing the right format? `electron-userland/electron-builder/pull/9279` implies it is now .icon

**Comment by claration** [comment]
> @khcrysalis I made the icon currently included with the idea that it would be for older macos versions while yours would be for macos 26+ with the new format. 
> 
> > This file will also work on lower versions, just so people know.
> 
> 
> 
> If it does work for lower versions & it's okay for the liquid glass style icon to be used on older macos ill swap it
> 
> 
> 
> Can you update the icon I tweaked the VC in the latest push. 
> 
> Also is the icns file you're sharing the right format? `electron-userland/electron-builder/pull/9279` implies it is now .icon

So here's the thing with `.icon` files. They're mainly used as a project file for the icon before fully compiling, when compiling Xcode takes in this file and spits out a `.icns` file (which is what I provided here). These are the files that macOS actually uses for these icons, Vesktop also uses a `.icns` file.

.icon =/= .icns

So we don't need to provide a `.icon` file unless we want to make tweaks from time-to-time to the icon. 

Apps have started adopting these icons recently and the new liquid glass appearances have started appearing on versions like Sequoia, it doesn't really effect much in anyway but it just shows you that they're choosing to move to that new design

**Comment by khcrysalis** [comment]
tldr: There's no new format in terms of macos icons, but there's a new way to make them. We only need the end product for Vesktop.

**Comment by claration** [comment]
After a bit more research, we need to bundle an additional file containing some more icon assets that I've missed
[Assets.car.zip](https://github.com/user-attachments/files/22628329/Assets.car.zip)
[vc.icns.zip](https://github.com/user-attachments/files/22607287/vc.icns.zip)

feel free to test.

**Comment by Vendicated** [comment]
We need to do this for the Assets. :car: (ignore steps 1 and 2) - https://github.com/electron-userland/electron-builder/issues/9254

image

**Comment by Vendicated** [comment]
I made the aforementioned changes but needs testing on Mac

**Comment by Vendicated** [comment]
it works but for some reason the images inside the Assets.car file are overly compressed and thus look terrible, so we have to fix that first

crunchiest 330bytes png ever

image

**Comment by Vendicated** [comment]
also we might be able to remove the .icns file entirely, will have to confirm with samsara

**Comment by Vendicated** [comment]
should be good to go now

**Comment by iblowmymind** [comment]
> also we might be able to remove the .icns file entirely, will have to confirm with samsara

Please correct me if I'm wrong, but I don't think older macOS systems can utilize icons inside Assets.car with the new Liquid Glass format so the icns should still be kept for compatibility with older macOS versions.

The current way icons are configured should be good; macOS <26 will use CFBundleIconFile to find the icon and macOS 26+ will look at CFBundleIconName to pick it up from the Assets.car file.

**Comment by Vendicated** [comment]
Assets.car is supported on older versions too so it should technically be enough

But if you don't specify an icns file, electron-builder will copy a default Electron icon in, so we won't remove it for now to avoid that

**Comment by claration** [comment]
@iblowmymind We've done a lot of testing on what macOS takes priority when it comes to icons using `CFBundleIconName` and `CFBundleIconFile`

https://developer.apple.com/documentation/bundleresources/information-property-list/cfbundleiconfile

https://developer.apple.com/documentation/bundleresources/information-property-list/cfbundleiconname

While yes `CFBundleIconFile` does technically support more macOS versions, Vesktop does not support below macOS Big Sur so this is not an issue for us. 

The Assets.car bundles extra assets for liquid glass on top of the supported assets for `CFBundleIconName`, so it can work between macOS versions without any issues.

From what we know Sequoia and Tahoe prioritises`CFBundleIconName` over the icon file, but for lower versions we don't exactly know so we're assuming it does the same there (lack of testing).

**Comment by iblowmymind** [comment]
Ah, understood, I thought versions below Tahoe simply ignored icons in Assets.car. Thanks a bunch!

**Comment by Red007Master** [comment]
Just came to say that this new icon Is "ugly ass"-iest that I have ever seen, and now I need to waste my time to fix it.

IDK why so many people have urge to:
1. Change Icon that is working.
2. Change it to ugly one.


Maybe somebody has path already written?
If somebody needs it: If I end up uploading patched version to AUR I will edit this comment.
